src/extras/global.js

```diff
@@ -10,18 +10,24 @@
   var firstGlobalProp, secondGlobalProp, lastGlobalProp;
   function getGlobalProp () {
     var cnt = 0;
-    var lastProp;
+    var foundLastProp, result;
     for (var p in global) {
       // do not check frames cause it could be removed during import
       if (shouldSkipProperty(p))
         continue;
       if (cnt === 0 && p !== firstGlobalProp || cnt === 1 && p !== secondGlobalProp)
         return p;
+      if (systemJSPrototype.firstGlobalProp) {
+        if (foundLastProp) {
+          lastGlobalProp = p;
+          result = result || p;
+        }
+        foundLastProp = foundLastProp || p === lastGlobalProp;
+      } else
+        result = p === lastGlobalProp ? result : p;
```

Comment:
This will detect existing globals if no global is defined I think? We should make sure to return `undefined` in the no global case. That's why I was under the impression we'd share your new `foundLastProp` logic between the two paths.

Reply:
[Your earlier code suggestion](https://github.com/systemjs/systemjs/pull/2280#discussion_r530070975) didn't work, so I changed it. But you are right, this has the problem where it sets the result before the lastGlobalProp is found. I'll update.